app/uk/gov/hmrc/tariffclassificationfrontend/controllers/CaseController.scala

```diff
@@ -45,10 +45,11 @@ class CaseController @Inject()(authenticatedAction: AuthenticatedAction,
                                val messagesApi: MessagesApi,
                                implicit val appConfig: AppConfig) extends FrontendController with I18nSupport {
 
+  private type Keyword = String
   private lazy val activityForm: Form[ActivityFormData] = ActivityForm.form
   private lazy val keywordForm: Form[String] = KeywordForm.form
 
-  def trader(reference: String): Action[AnyContent] = authenticatedAction.async { implicit request =>
+  def trader(reference: String): Action[AnyContent] = verify.caseExistsAndCheckPermissions(reference).async { implicit request =>
     getCaseAndRenderView(
```

Comment:
If we have the case on the request we dont need the `getCase` call any more in the controller